clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -587,6 +599,18 @@ public boolean shouldSkipHeartbeat() {
         return state == MemberState.UNSUBSCRIBED || state == MemberState.FATAL;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void transitionToStaled() {
+        transitionTo(MemberState.PREPARE_LEAVING);
+        memberEpoch = ConsumerGroupHeartbeatRequest.LEAVE_GROUP_MEMBER_EPOCH;
+        transitionTo(MemberState.LEAVING);
+        leaveGroupInProgress = Optional.of(CompletableFuture.completedFuture(null));
+        transitionTo(MemberState.STALED);
```

Comment:
We also need to make sure that when going down this path, we clear the current assignment, so the member actually rejoins with its subscription, but ready to take on whatever new partitions the broker assigns.

Reply:
nvm - i was confused by the pending assignments.  I think we need to clear the current assignment.